### PR TITLE
Use GA topology labels for Azuredisk

### LINKED ISSUE #98281: Update in-tree PV kubernetes topology label to GA
**What happened**:
The current in-tree PV topology is still using beta version label "failure-domain.beta.kubernetes.io". This was intentionally not upgrade because of CSI migration. https://github.com/kubernetes/kubernetes/pull/81431#discussion_r341216734

However, since the CSI migration timeline is likely to slip, and with the fix of https://github.com/kubernetes/kubernetes/pull/97823 , it should be safe to update the in-tree PVC to GA version (`topology.kubernetes.io`) So that we can unblock removing the beta label from node.

**What you expected to happen**:
Replace all beta topology label to GA version once https://github.com/kubernetes/kubernetes/pull/97823 merged.

/sig storage
/cc @msau42 @verult @kassarl

### PR DESCRIPTION
**Pull request by kassarl**
#### What type of PR is this?

/kind feature
#### What this PR does / why we need it:
Similar to #98700, this PR replaces the deprecated topology labels in favor of the new GA ones for Azure disk

#### Which issue(s) this PR fixes:

Fixes #98281

#### Does this PR introduce a user-facing change?

```release-note
[ACTION REQUIRED] Newly provisioned PVs by Azure disk will no longer have the beta FailureDomain label. Azure disk volume plugin will start to have GA topology label instead.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by Jiawei0227** [comment]
Can you also delete the extra line in release notes?
/assign

/cc @andyzhangx it seems the aks test errors: "Cannot create more than 100 public IP addresses for this subscription in this region." Can you help take a look?

**Review by andyzhangx — APPROVED** [review]
/lgtm
/approve

**Comment by Jiawei0227** [comment]
/lgtm
@msau42 for csi-translation-lib approval

File: staging/src/k8s.io/csi-translation-lib/plugins/azure_disk_test.go [hunk 1]
```diff
@@ -269,7 +269,7 @@ func TestTranslateInTreeStorageClassToCSI(t *testing.T) {
 		},
 		{
 			name:       "some translated topology",
-			options:    NewStorageClass(map[string]string{}, generateToplogySelectors(v1.LabelFailureDomainBetaZone, []string{"foo"})),
+			options:    NewStorageClass(map[string]string{}, generateToplogySelectors(v1.LabelTopologyZone, []string{"foo"})),
```
Comment (msau42) [inline staging/src/k8s.io/csi-translation-lib/plugins/azure_disk_test.go:272]: Can we keep the test case for the beta labels too? This function needs to support both beta and GA labels.

**Review by andyzhangx — APPROVED** [review]
/lgtm
/approve
@andrewsykim could you approve? thanks.

**Review by andrewsykim — COMMENTED** [review]
/approve

(just approving the cloud provider changes since this is necessary with the old beta labels being deprecated, I have not extensively reviewed the PV logic)

**Pull request merged**